doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,106 +1,930 @@
+                                        ; (i = i0) → ouc u0 })
+                               (ouc u0)
+
+When ``i`` is ``i0`` this is ``u0`` and when ``i`` is ``i1`` this is
+``hcomp u u0``. This can hence be seen as giving us the interior of an
+open box. In the special case of the square above ``hfill`` gives us a
+direct cubical proof that composing ``p`` with ``refl`` is ``p``.
+
+::
+
+  compPathRefl : ∀ {ℓ} {A : Set ℓ} {x y : A} (p : x ≡ y) → compPath p refl ≡ p
+  compPathRefl {x = x} {y = y} p j i = hfill (λ _ → λ { (i = i0) → x
+                                                      ; (i = i1) → y })
+                                             (inc (p i))
+                                             (~ j)
+
 
+Glue types
 ----------
+
+In order to be able to prove the univalence theorem we also have to
+add "Glue" types. These lets us turn equivalences between types into
+paths between types. An equivalence of types ``A`` and ``B`` is
+defined as a map ``f : A → B`` such that its fibers are contractible.
+
+.. code-block:: agda
+
+  fiber : ∀ {ℓ} {A B : Set ℓ} (f : A → B) (y : B) → Set ℓ
+  fiber {A = A} f y = Σ[ x ∈ A ] f x ≡ y
+
+  isContr : ∀ {ℓ} → Set ℓ → Set ℓ
+  isContr A = Σ[ x ∈ A ] (∀ y → x ≡ y)
+
+  record isEquiv {ℓ} {A B : Set ℓ} (f : A → B) : Set ℓ where
+    field
+      equiv-proof : (y : B) → isContr (fiber f y)
+
+  _≃_ : ∀ {ℓ} (A B : Set ℓ) → Set ℓ
+  A ≃ B = Σ[ f ∈ (A → B) ] (isEquiv f)
+
+The simplest example of an equivalence is the identity function.
+
+::
+
+  idfun : ∀ {ℓ} → (A : Set ℓ) → A → A
+  idfun _ x = x
+
+  idIsEquiv : ∀ {ℓ} (A : Set ℓ) → isEquiv (idfun A)
+  equiv-proof (idIsEquiv A) y =
+    ((y , refl) , λ z i → z .snd (~ i) , λ j → z .snd (~ i ∨ j))
+
+  idEquiv : ∀ {ℓ} (A : Set ℓ) → A ≃ A
+  idEquiv A = (idfun A , idIsEquiv A)
+
+
+An important special case of equivalent types are isomorphic types
+(i.e. types with maps going back and forth which are mutually
+inverse): https://github.com/agda/cubical/blob/master/Cubical/Foundations/Isomorphism.agda.
+
+As everything has to work up to higher dimensions the Glue types take
+a partial family of types that are equivalent to the base type ``A``:
+
+::
+
+  Glue : ∀ {ℓ ℓ'} (A : Set ℓ) {φ : I}
+       → Partial φ (Σ[ T ∈ Set ℓ' ] T ≃ A) → Set ℓ'
+
+..
+  ::
+
+  Glue A Te = primGlue A (λ x → Te x .fst) (λ x → Te x .snd)
+
+These come with a constructor and eliminator:
+
+.. code-block:: agda
+
+  glue : ∀ {ℓ ℓ'} {A : Set ℓ} {φ : I} {Te : Partial φ (Σ[ T ∈ Set ℓ' ] T ≃ A)}
+       → PartialP φ T → A → Glue A Te
+
+  unglue : ∀ {ℓ ℓ'} {A : Set ℓ} (φ : I) {Te : Partial φ (Σ[ T ∈ Set ℓ' ] T ≃ A)}
+         → Glue A Te → A
+
+Using Glue types we can turn an equivalence of types into a path as
+follows:
+
+::
+
+  ua : ∀ {ℓ} {A B : Set ℓ} → A ≃ B → A ≡ B
+  ua {_} {A} {B} e i = Glue B (λ { (i = i0) → (A , e)
+                                 ; (i = i1) → (B , idEquiv B) })
+
+The idea is that we glue on ``A`` to ``B`` when ``i = i0`` using ``e``
```

Comment:
"we glue on A to B" seems grammatically weird, I had to read it several times to understand what it means. Is "on" needed at all? Maybe it should be "glue A on to B"?

Reply:
I pushed "The idea is that we glue ``A`` together with ``B`` when ``i = i0`` using ``e`` and ``B`` with itself when ``i = i1`` using the identity equivalence."

Is it clearer?